src/compiler/checker.ts

```diff
@@ -4333,6 +4338,12 @@ module ts {
             return !!getPropertyOfType(type, "0");
         }
 
+        function isTupleType(type: Type) : boolean {
+            // Check if a Type exactly implements interface TupleType. Typical typechecking code should rely on

```

Comment:
There isn't actually an interface named `TupleType`. It's more accurate to say that the type was explicitly written as a [tuple type literal](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.7.5).

Additionally, could you move this comment above `isTupleType` and convert it into a multiline doc comment like the following?

```TypeScript
/**
 * Lorem ipsum etc. etc.
 */
```